t/06elements.t

```diff
@@ -0,0 +1,605 @@
+# $Id$
+
+##
+# this test checks the DOM element and attribute interface of XML::LibXML
+
+use v6.c;
+use Test;
+
+#use base qw(ParentClass);
+#
+#sub new {
+#    my $class = shift;
+#    $class = ref $class if ref $class;
+#    my $self = bless {}, $class;
+#    $self;
+#}
+
+#1;
+
+# Should be 187.
+#use Test::More tests => 191;
+plan 187;
+
+
+use XML::LibXML;
+
+my $foo       = "foo";
+my $bar       = "bar";
+my $nsURI     = "http://foo";
+my $prefix    = "x";
+my $attname1  = "A";
+my $attvalue1 = "a";
+my $attname2  = "B";
+my $attvalue2 = "b";
+my $attname3  = "C";
+
+# TEST:$badnames=4;
+my @badnames= ("1A", "<><", "&", "-:");
+
+# 1. bound node
+{
+    my $doc = XML::LibXML::Document.new();
+    my $elem = $doc.createElement( $foo );
+    # TEST
+    ok $elem, 'created element';
+    # TEST
+    is $elem.tagName, $foo, 'element has correct tag name';
+
+    {
+        for @badnames -> $name {
+            try { 
+                my $caught = 0;
+                CATCH {
+                    ok $@, "setNodeName throws an exception for $name";
```

Comment:
Actually, that is why I've written the block in this way (and why in the .pm files you will find a return after a die(). If there is an exception, $caught will never be set and the nok (soon to be fail) will be the only test. If it succeeds, the ok (soon to be pass) is done and the nok/fail will not execute due to the unless clause. You don't see them in this file because I fixed those in t/06elements-port.t see [(https://github.com/FROGGS/p6-XML-LibXML/pull/1/files/223ef068513e236e2b6df64d91fe75243da87973#diff-c8808174d37f2722a60f4c71e878aacaR68)]